code/modules/events/ghost_role.dm

```diff
@@ -8,18 +8,19 @@
 /datum/round_event/ghost_role/start()
 	try_spawning()
 
-/datum/round_event/ghost_role/proc/try_spawning(sanity = 0)
+/datum/round_event/ghost_role/proc/try_spawning(sanity = 0, retry = 0)
 	// The event does not run until the spawning has been attempted
 	// to prevent us from getting gc'd halfway through
 	processing = FALSE
 
 	var/status = spawn_role()
-	if(status == WAITING_FOR_SOMETHING)
+	if((status == WAITING_FOR_SOMETHING) && (retry < 3))
```

Comment:
If this retries the maximum amount of times, i think it'll throw the error on line 37(even though it's not a bug); it should properly handle excessive retries and refund the event creation so it can possibly try again later.